packages/commons/src/eservice-documents/eserviceDocumentUtils.ts

```diff
@@ -310,6 +311,15 @@ export const extractEServiceUrlsFrom = async (
       .otherwise(() => {
         throw new Error();
       });
+
+    // Validate that all returned strings are valid URLs
+    for (const url of serverUrls) {
+      if (!URL.canParse(url)) {
```

Comment:
`z.string().url()` is the same method that uses the data response validator. Why not use it?
```suggestion
      try {
         z.string().url().parse(url);
      } catch {
        throw invalidServerUrl(resourceId); 
      }
```